src/actions/submit/validate_branches.ts

```
@@ -8,20 +8,29 @@ export async function validateBranchesToSubmit(
   branchNames: string[],
   context: TContext
 ): Promise<string[]> {
-  await Promise.all([
-    validateNoMergedOrClosedBranches(branchNames, context),
-    validateNoEmptyBranches(branchNames, context),
-    validateBaseRevisions(branchNames, context),
-  ]);
+  const syncPrInfoPromise = syncPrInfo(branchNames, context);
```

What we really wanted here was to fetch the GH data while we're doing other stuff — this truly gives us that — promise.all is overkill and breaks

---

What ended up being the reason `Promise.all` would break and how this fixes it?

(Specifically, I see here that if any of the methods end up throwing their errors will still bubble up so I'm not sure how we solved the original problem.)

---

because we could throw an exception after the promise.all was resolved